.github/workflows/cypress-e2e.yml

```diff
@@ -233,37 +229,37 @@ jobs:
           ./occ collectives:create SearchTest --owner=bob
           ./occ collectives:index
 
+      - name: Register index for cypress-split
+        env:
+          container: "${{ matrix.containers }}"
+        run: |
+          echo "container_index: $(($container - 1))"
+          echo "container_index=$(($container - 1))" >> $GITHUB_ENV
+
       - name: Run E2E cypress tests
-        uses: cypress-io/github-action@8d3918616d8ac34caa2b49afc8b408b6a872a6f5 # v6.7.1
-        with:
-          record: '${{ !!matrix.use-cypress-cloud }}'
-          parallel: '${{ !!matrix.use-cypress-cloud }}'
-          group: "${{ !!matrix.use-cypress-cloud && format('Nextcloud {0}',  matrix.server-versions) || '' }}"
-          wait-on: '${{ env.CYPRESS_baseUrl }}'
-          working-directory: apps/${{ env.APP_NAME }}
-          config: video=false,defaultCommandTimeout=20000
-          tag: "${{ !!matrix.use-cypress-cloud && github.event_name || '' }}"
+        run: |
+          cd 'apps/${{ env.APP_NAME }}'
+          npx wait-on $CYPRESS_baseUrl
+          npx cypress run --record false --config defaultCommandTimeout=20000,video=false
         env:
-          # https://github.com/cypress-io/github-action/issues/124
-          COMMIT_INFO_MESSAGE: ${{ github.event.pull_request.title }}
-          COMMIT_INFO_SHA:  ${{ github.event.pull_request.head.sha }}
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
           CYPRESS_ncVersion: ${{ matrix.server-versions }}
+          # `SPLIT` needs to match count of `containers` in the matrix above
+          SPLIT: 4
+          SPLIT_INDEX: ${{ env.container_index }}
 
       - name: Upload snapshots
         uses: actions/upload-artifact@0b2256b8c012f0828dc542b3febcab082c67f72b # v4.3.4
         if: failure()
         with:
-          name: Screenshots.${{ matrix.server-versions }}-${{ matrix.containers }}
+          name: screenshots_${{ matrix.server-versions }}_${{ matrix.containers }}
           path: apps/${{ env.APP_NAME }}/cypress/screenshots/
           retention-days: 5
 
       - name: Upload NC logs
         uses: actions/upload-artifact@0b2256b8c012f0828dc542b3febcab082c67f72b # v4.3.4
-        if: always()
+        if: failure()
```

Comment:
Is there a disadvantage to always uploading them?

Most of the time I think the logs should be empty, right? Sometimes i found it useful to check for that and investigate errors in the logs even if the tests passed.

Reply:
I changed this because I never look into logs when tests succeed, and it's like this in Text. Don't have strong feelings about it though :shrug: